src/components/Executions/ExecutionFilters.tsx

```diff
@@ -57,21 +91,41 @@ export const ExecutionFilters: React.FC<{ filters: FilterState[] }> = ({
     filters
 }) => {
     const styles = useStyles();
+    const { profile } = React.useContext(APIContext);
+
     return (
         <div className={styles.container}>
             {filters.map(filter => {
-                const renderContent = () => <RenderFilter filter={filter} />;
-                return (
-                    <FilterPopoverButton
-                        {...filter.button}
-                        active={filter.active}
-                        key={filter.label}
-                        onReset={filter.onReset}
-                        buttonText={filter.label}
-                        className={styles.filterButton}
-                        renderContent={renderContent}
-                    />
-                );
+                if (filter.type == 'byUser') {
```

Comment:
* I think a Checkbox might make more sense for this interaction.
* Preference for `===` instead of `==`. We want strict equality for string checks.
* If we add the `hidden` field to filter state, we can simplify this a little bit.
* The preference in this code base is to return `null` instead of an empty fragment(`<></>`) so that React knows there is no work to do.

Those changes might look something like:
```tsx
{filters.map(filter => {
    if ( !!filter.hidden ) {
        return null;
    }
    if (filter.type === 'boolean') {
        const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => filter.setActive(event.target.checked);
        // Using a standard MaterialUI checkbox component. You may want to do this differently
        return (
            <FormControlLabel
                key={filter.label}
                control={
                    <Checkbox
                        checked={filter.active}
                        onChange={handleChange}
                     />
                }
                label={filter.label}
        />)
    }
    const renderContent = () => <RenderFilter filter={filter} />;
    return (
        <FilterPopoverButton
            {...filter.button}
            active={filter.active}
            key={filter.label}
            onReset={filter.onReset}
            buttonText={filter.label}
            className={styles.filterButton}
            renderContent={renderContent}
        />
    );
}
```